### ISSUE TITLE
Windows Hello prompt does not appear after hibernate

### ISSUE THREAD

**Issue by twistios** [keepassxreboot/keepassxc#8831]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
After my system wakes up from hibernate the database is locked. To unlock it, it expects me to type in my password using a Windows Hello dialog. This dialog does not always appear so I am not able to unlock the database without restarting KeePassXC.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. open KeePassXC and unlock a database
2. set Windows into hibernate mode (probably energy saving mode also works)
3. wait some time
4. wake up the system and login
5. unlock the database by clicking on "unlock database"

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
The Windows Hello dialog should appear requesting the Windows account pin (password).

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
The Hello dialog does not appear. After exiting the program and starting it again the Hello dialog appears (now useless). Only after canceling all Hello dialogs (sometimes multiple appear) I am able to start KeePassXC (using another click on the icon).

## Context
[NOTE]: # ( Give us any additional information you may have. )
This problem is not happening every time, sometimes it works like expected and sometimes it asks again for the database passphrase.
I normally have two databases open at once. My database uses Windows Hello as an extra security, I don't know if this is a default setting. The database was originally created using KeePass 2.

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.4
Revision: 63b2394

Qt 5.15.6
Diagnosemodus ist deaktiviert.

Betriebssystem: Windows 10 Version 2009
CPU-Architektur: x86_64
Kernel: winnt 10.0.19044

Aktivierte Erweiterungen:
- Auto-Type
- Browser-Integration
- SSH-Agent
- KeeShare
- YubiKey
- Schnelle Entsperrung

Kryptographische Bibliotheken:
- Botan 2.19.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows 10 Version 21H2

**Comment by JohnLGalt** [comment]
I've noticed similar behavior - if the Windows Hello prompt is about to load, and you click ***anywhere*** on screen / on app during the WH load process, instead of authenticating when you press OK on the WH prompt, it continually tries to reload and have you press OK multiple times - I usually have to keep clicking on OK 4-5 times before it finally accepts the OK button press.

However, I've also noticed that WH oftentimes will not load in front of all other windows, which is how I fall into the above situation in the first place. Sometimes the unlock process works exactly as expected, and sometimes it just continually fails to authenticate and just pops right back up again after clicking the OK button, and usually this is because there are other windows open, and clicking on even on to minimize it results in the WH dialog losing its focus (or mind).

**Comment by droidmonkey** [comment]
This is 100% because the WH dialog is behind other windows. If your computer is slow or processing a lot of other things then it will take time for the window to appear. This is NOT a keepassxc problem though. Since 2.7.3, we seek to bring the WH dialog to the top of the stack multiple times. Just be on the lookout for a new window in your Taskbar, that is the WH dialog.

**Comment by JohnLGalt** [comment]
Even if the window is on top, though, if you click anywhere else _before_ clicking the OK button, even accidentally, it requires multiple attempts to authenticate as clicking the OK button just brings the dialog (and attempted authentication) back again - and it takes several more attempts to actually get it to accept it. If I click really, really quickly in the same spot where the OK button is, I an usually get it to accept the OK button after the 4th iteration of button click --> WH dialog popping back up.

Granted, I don't think this is a KPXC bug, but rather, a WH bug in and of itself. Still, though, annoying, when it happens.

And as for my computer - it may not be curent gen, but it is **definitely _not_ slow**.

I only have one other thing that actually brings up WH, and I have yet to duplicate this issue with that. I wonder if the attempts to bring it to the front are part of the cause of WH going stupid?

I also wish Windows would enable WH on Secure Desktop, that, I think, would solve a lot of problems in and of itself.

**Comment by twistios** [comment]
The icon of the windows hello prompt does only appear in the Taskbar (in the described case) once I closed KeePassXC **after** I clicked on the KeePassXC icon on the desktop but then directly (often multiple once, maybe because I clicked on the "unlock"-button multiple times).

**Comment by droidmonkey** [comment]
Can you rephrase the above because it doesn't make sense to me.

**Comment by twistios** [comment]
In the problematic case:
1. I click the unlock button
2. The wheel on the mouse pointer spins indicating to load something
3. Nothing happens so I close KeePassXC
4. I click on the icon of KeePassXC on my desktop
5. The Windows Hello prompt(s) appear, as a window and as an icon on the taskbar
6. I cancel all of them
7. I click again on the icon of the program
8. Everything is fine now.
